src/utf8lex_definition.c

```diff
@@ -312,6 +312,7 @@ utf8lex_error_t utf8lex_printable_str(
     if (new_target_bytes >= max_bytes)
     {
       // Can't fit the printable characters plus \0 into the target.
+      printable_str[target_offset] = '\0';
```

Comment:
When the unit tests started running on mips64le and ppc64le, it became clear that we were not terminating printable_str in the case where we exit early with `UTF8LEX_MORE`.  On other architectures, the bytes were all 0 to begin with, so we did not detect the unterminated string.  The test_printable_str.c unit test was the one that failed.